### PR TITLE
[13.0][FIX] sale_line_refund_to_invoice_qty, fix default value on aml

### PR DESCRIPTION
**Pull request by ChrisOForgeFlow**
[BP] #1381 

@ForgeFlow

cc: @JordiMForgeFlow

### CODE REVIEW

File: sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            (reference_date,),
+        )
+        env.cr.commit()
+        query = """
+        with invoice_lines as (
+        select rel.order_line_id, sum(coalesce(aml.quantity, 0)) as quantity
+        from account_move_line as aml
+        inner join account_move as am on am.id = aml.move_id
+        inner join sale_order_line_invoice_rel as rel on rel.invoice_line_id  = aml.id
+        where am.state = 'posted' and am.move_type = 'out_invoice'
```
Comment (JordiMForgeFlow) [inline sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py]:
```suggestion
        where am.state = 'posted' and am.type = 'out_invoice'
```

File: sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        group by rel.order_line_id
+        ), credit_lines as (
+            select rel.order_line_id, sum(coalesce(aml.quantity, 0)) as quantity
+            from account_move_line as aml
+            inner join account_move as am on am.id = aml.move_id
+            inner join sale_order_line_invoice_rel as rel
+            on rel.invoice_line_id  = aml.id
+            where am.state = 'posted' and am.move_type = 'out_refund'
```
Comment (JordiMForgeFlow) [inline sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py]:
```suggestion
            where am.state = 'posted' and am.type = 'out_refund'
```

File: sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        """
+        env.cr.execute(query, (reference_date, reference_date))
+        lines_with_problems = [data[0] for data in env.cr.fetchall()]
+        if lines_with_problems:
+            _logger.debug("total lines to reprocess: %s", str(len(lines_with_problems)))
+            sol_model.search(
+                [
+                    ("id", "in", lines_with_problems),
+                    ("product_id.type", "!=", "service"),
```
Comment (JordiMForgeFlow) [inline sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py]: Maybe we could already include the filtering of service products in the query?

Comment (ChrisOForgeFlow) [reply sale_line_refund_to_invoice_qty/migrations/13.0.2.2.1/post-fix-old-field-value.py]: Done!

**Comment by JordiMForgeFlow** [comment]
Hi @ChrisOForgeFlow 

Changes look good! I think that the migration script file should be called post-migration as a convention.

@MiquelRForgeFlow could you review also?

File: sale_line_refund_to_invoice_qty/__manifest__.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "name": "Sale Line Refund To Invoice Qty",
     "summary": """Allow deciding whether refunded quantity should be considered
                 as quantity to reinvoice""",
-    "version": "13.0.1.2.1",
+    "version": "13.0.2.2.1",
```
Comment (MiquelRForgeFlow) [inline sale_line_refund_to_invoice_qty/__manifest__.py]:
```suggestion
    "version": "13.0.2.0.0",
```
why not this?

**Review by MiquelRForgeFlow — COMMENTED** [review]
@JordiMForgeFlow it's not important the name. It beginning as `post-` it's the only thing that really matters.

File: sale_line_refund_to_invoice_qty/__manifest__.py (hunk 4 above)
Comment (ChrisOForgeFlow) [reply sale_line_refund_to_invoice_qty/__manifest__.py]: Done!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request closed without merging**